src/com/facebook/buck/jvm/common/ResourceValidator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2014-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.facebook.buck.jvm.common;
+
+import com.facebook.buck.io.ProjectFilesystem;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.util.HumanReadableException;
+import com.google.common.collect.ImmutableSortedSet;
+
+import java.nio.file.Path;
+
+public final class ResourceValidator {
+  // TODO(natthu): Consider adding a validateArg() method on Description which gets called before
+  // createBuildable().
```

Comment:
I don't think this `TODO` makes much sense anymore now that we've moved it, so feel free to nuke it.